File: src/main/java/at/jku/ssw/coco/intellij/CocoRefactoringProvider.java
package at.jku.ssw.coco.intellij;

import at.jku.ssw.coco.intellij.psi.CocoNamedElement;
import com.intellij.lang.refactoring.RefactoringSupportProvider;
import com.intellij.psi.PsiElement;

/**
 * Created by Thomas on 29/03/2015.
 */
public class CocoRefactoringProvider extends RefactoringSupportProvider {
    @Override
    public boolean isMemberInplaceRenameAvailable(PsiElement element, PsiElement context) {
        return element instanceof CocoNamedElement;
    }
}

File: src/main/kotlin/me/salzinger/intellij/coco/CocoRefactoringProvider.kt
package me.salzinger.intellij.coco

import com.intellij.lang.refactoring.RefactoringSupportProvider
import com.intellij.psi.PsiElement
import me.salzinger.intellij.coco.psi.CocoNamedElement

/**
 * Created by Thomas on 29/03/2015.
 */
class CocoRefactoringProvider : RefactoringSupportProvider() {
    override fun isMemberInplaceRenameAvailable(element: PsiElement, context: PsiElement?): Boolean {
        return element is CocoNamedElement
    }
}
